### PR TITLE
Implement voice mute and migration from voice ban -> voice mute

### LINKED ISSUE #605: Add new infraction type to backend
In regards to python-discord/bot#1849

The linked issue will do the following:

- Rename the current `voiceban` command to `voicemute`
- Create a new command called `voiceban`

So we will need to add a relevant new type as well as convert the current `voiceban` infractions to `voicemute`s.

### PR DESCRIPTION
**Pull request by ks129**
Closes #605 

Added new `voice_mute` infraction type, added migration to migrate existing voice bans to voice mutes. Had to upper type's max length from 9 to 10.

Voice mute cannot be hidden.

**THIS HAS TO BE MERGED AT THE SAME TIME AS BOT PR!**

### CODE REVIEW

File: pydis_site/apps/api/migrations/0074_voice_mute.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        infraction.type = "voice_mute"
+        infraction.save()
+
+
+def unmigrate_infractions(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Infraction = apps.get_model("api", "Infraction")
+
+    for infraction in Infraction.objects.filter(type="voice_mute"):
+        infraction.type = "voice_ban"
+        infraction.save()
```
Comment (jchristgit) [inline pydis_site/apps/api/migrations/0074_voice_mute.py:20]: Hmm, out of curiosity, can't Django do a bulk update for all these infractions? It's probably a small count anyways, so it doesn't matter too much.

**Review by jchristgit — APPROVED** [review]
This looks good to me. Thank you!

**Comment by jchristgit** [comment]
> THIS HAS TO BE MERGED AT THE SAME TIME AS BOT PR!

Can you link the bot PR? From a quick search, I couldn't find it

**Comment by onerandomusername** [comment]
Might be worth turning off automerge since it needs to be merged at the same time as the bot pr....

**Comment by ChrisLovering** [comment]
Hey @ks129 when you get some time, could you resolve the multiple leaf nodes issue in the migrations? I merged in main so that I could test this against the bot's main branch, but then got some migration conflicts :D

**Review by ChrisLovering — APPROVED** [review]
Yup, works well! Tested with the bot side (https://github.com/python-discord/bot/pull/2052) and infractions get saved, rescheduled etc and all looks good, thanks! 👍 

Added the do not merge label since this needs to be merged at the same time as the bot side.

**Pull request merged**